core/indexer/dataDispatcher.go

```diff
@@ -74,12 +118,37 @@ func (d *dataDispatcher) Add(item workItems.WorkItemHandler) {
 		log.Warn("dataDispatcher.Add nil item: will do nothing")
 		return
 	}
+	if d.wasClosed.IsSet() {
+		log.Warn("dataDispatcher.Add cannot add item: channel chanWorkItems is closed")
+		return
+	}
 
 	d.chanWorkItems <- item
+
+	if d.wasClosed.IsSet() {
+		close(d.chanWorkItems)
+	}
 }
 
-func (d *dataDispatcher) doWork(wi workItems.WorkItemHandler) {
+func (d *dataDispatcher) doWork(wi workItems.WorkItemHandler, timeout time.Duration) {
+	if wi == nil {
```

Comment:
since we won't close the channel, this check is useless now

Reply:
do we still need this nil check?

Reply:
removed.